packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/XmlDocExceptionStatement.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Generator.CSharp.Providers;
+
+namespace Microsoft.Generator.CSharp.Statements
+{
+    public class XmlDocExceptionStatement : MethodBodyStatement
+    {
+        //($"<exception cref=\"{ExceptionType}\">", "</exception>", Text)
+        public Type ExceptionType { get; }
+        public IReadOnlyList<ParameterProvider> Parameters { get; }
+
+        private string _reason;
+
+        public XmlDocExceptionStatement(ParameterValidationType validationType, IReadOnlyList<ParameterProvider> parameters)
+        {
+            ExceptionType = GetExceptionType(validationType);
+            Parameters = parameters;
+            _reason = GetText(validationType);
+        }
+
+        private static Type GetExceptionType(ParameterValidationType validationType) => validationType switch
+        {
+            ParameterValidationType.AssertNotNull => typeof(ArgumentNullException),
+            ParameterValidationType.AssertNotNullOrEmpty => typeof(ArgumentException),
+            _ => throw new ArgumentOutOfRangeException(nameof(validationType), validationType, $"Cannot create an XmlDocExceptionStatement with {validationType} validationType")
+        };
+
+        private static string GetText(ParameterValidationType validationType) => validationType switch
+        {
+            ParameterValidationType.AssertNotNull => "is null.",
+            ParameterValidationType.AssertNotNullOrEmpty => "is an empty string, and was expected to be non-empty.",
+            _ => throw new ArgumentOutOfRangeException(nameof(validationType), validationType, $"Cannot create an XmlDocExceptionStatement with {validationType} validationType")
+        };
+
+        internal override void Write(CodeWriter writer)
+        {
+            writer.WritingXmlDocumentation = true;
```

Comment:
I wonder if this could be implemented using something like an XmlWritingScope that implements IDisposable.
```c#

internal class XmlWritingScope : IDisposable 
{
   private readonly CodeWriter _writer;
   public XmlWritingScope(CodeWriter writer)
   {
     _writer = writer;
     _writer .WritingXmlDocumentation = true;
   }

  public void Dispose()
  {
      _writer.WritingXmlDocumentation = false;
  }
```

I guess it would end up being more lines total but at least we abstract away the property.